include/alpaka/mem/view/Traits.hpp

```diff
@@ -353,4 +378,100 @@ namespace alpaka
     {
         return traits::CreateStaticDevMemView<TDev>::createStaticDevMemView(pMem, dev, extent);
     }
+
+    //! Creates a view to a pointer device
+    //!
+    //! \param pMem Pointer to memory.
+    //! \param dev Device from where pMem can be accessed.
+    //! \param extent Number of elements represented by the pMem.
+    //!               Using a multi dimensional extent will result in a multi dimension view to the memory represented
+    //!               by pMem.
+    //! \return A view to device memory.
+    template<typename TElem, typename TDev, typename TExtent>
+    auto createView(TElem* pMem, TDev const& dev, TExtent const& extent)
+    {
+        return traits::CreateViewPlainPtr<TDev>::createViewPlainPtr(
+            pMem,
+            dev,
+            extent,
+            detail::calculatePitchesFromExtents<TElem>(extent));
+    }
+
+    //! Creates a view to a device pointer
+    //!
+    //! \param pMem Pointer to memory.
+    //! \param dev Device from where pMem can be accessed.
+    //! \param extent Number of elements represented by the pMem.
+    //!               Using a multi dimensional extent will result in a multi dimension view to the memory represented
+    //!               by pMem.
+    //! \param pitch Pitch in bytes for each dimension. Dimensionality must be equal to extent.
+    //! \return A view to device memory.
+    template<typename TElem, typename TDev, typename TExtent, typename TPitch>
+    auto createView(TElem* pMem, TDev const& dev, TExtent const& extent, TPitch const& pitch)
+    {
+        return traits::CreateViewPlainPtr<TDev>::createViewPlainPtr(pMem, dev, extent, pitch);
+    }
+
+    //! Creates a view to a device std::vector
+    //!
+    //! \param vec Pointer to memory.
+    //! \param dev Device from where vec can be accessed.
+    //! \return A view to device memory.
+    template<typename TElem, typename TAllocator, typename TDev>
+    auto createView(std::vector<TElem, TAllocator>& vec, TDev const& dev)
+    {
+        return createView(vec.data(), dev, extent::getExtent(vec));
```

Comment:
Not for this PR, since it would require C++17. But theoretically [std::data](https://en.cppreference.com/w/cpp/iterator/data) would be nice to unify all contiguous containers.